### ISSUE TITLE
Error while slicing PaginatedList with few items

### ISSUE THREAD

**Issue by citaret** [linode/python-linode-api#39]
This will not happen while there are hundreds of items.

```
In [230]: g1 = client.linode.get_instances(linode.Linode.group == "xxx")

In [231]: g1[0]
Out[231]: Linode: 3524678

In [232]: g1[1]
Out[232]: Linode: 3524954

In [233]: g1[:]
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-233-528aa4d21326> in <module>()
----> 1 g1[:]

~/.virtualenvs/vpn3/lib/python3.6/site-packages/linode/util/paginated_list.py in __getitem__(self, index)
     47         # this comes in here now, but we're hadling it elsewhere
     48         if isinstance(index, slice):
---> 49             return self._get_slice(index)
     50
     51         # handle negative indexing

~/.virtualenvs/vpn3/lib/python3.6/site-packages/linode/util/paginated_list.py in _get_slice(self, s)
    106         if not self.lists[i_page]:
    107             self._load_page(i_page)
--> 108         if not self.lists[j_page]:
    109             self._load_page(j_page)
    110

IndexError: list index out of range
```

**Comment by Dorthu** [comment]
Hmm.. it appears that taking a full slice of the PaginatedList raises.  I'll look into, thanks for reporting.

**Comment by citaret** [comment]
Seems that it's related to https://github.com/linode/python-linode-api/issues/42

**Comment by Dorthu** [comment]
I think I need to rewrite the slicing code, because it was pretty buggy around the edge cases.  Was this issue resolved?

**Comment by citaret** [comment]
I tested the issue in the version 4.1.3b0, it exists in the upper bound.

```
In [1]: import linode

In [2]: client = linode.LinodeClient('unauthenticated')

In [3]: group = client.linode.get_kernels(linode.Kernel.architecture==True)

In [4]: x = group[:-1]

In [5]: y = group[:]
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-5-67730774c4d6> in <module>()
----> 1 y = group[:]

~/.virtualenvs/vpn3/lib/python3.6/site-packages/linode/paginated_list.py in __getitem__(self, index)
     47         # this comes in here now, but we're hadling it elsewhere
     48         if isinstance(index, slice):
---> 49             return self._get_slice(index)
     50
     51         # handle negative indexing

~/.virtualenvs/vpn3/lib/python3.6/site-packages/linode/paginated_list.py in _get_slice(self, s)
    106         if not self.lists[i_page]:
    107             self._load_page(i_page)
--> 108         if not self.lists[j_page]:
    109             self._load_page(j_page)
    110

IndexError: list index out of range
```

**Comment by Dorthu** [comment]
This fix has been released in v4.1.3b1
